Samples/Notifications/App/CppUnpackagedAppNotifications/CppUnpackagedAppNotifications/App.xaml.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include "pch.h"
+#include "App.xaml.h"
+#include "MainWindow.xaml.h"
+#include "Notifications\NotificationManager.h"
+#include <wil/result.h>
+#include <Microsoft.UI.Xaml.Window.h>
+#include <winrt/Microsoft.Windows.AppLifecycle.h>
+#include <winrt/Microsoft.Windows.AppNotifications.h>
+
+namespace winrt
+{
+    using namespace Windows::Foundation;
+    using namespace Microsoft::UI::Xaml;
+    using namespace winrt::Microsoft::Windows::AppLifecycle;
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+}
+
+// NotificationManager is responsible for registering and unregistering the Sample for App Notifications as well as
+// dispatching actionned notifications to the appropriate scenario.
+// Registration will happen when Init() is called and Unregistration will happen when this
+// instance variable goes out of scope, i.e.: when the App is terminated.
+static NotificationManager g_notificationManager;
+
+namespace winrt::CppUnpackagedAppNotifications::implementation
+{
+    static App* app{ nullptr };
+
+    App::App()
+    {
+        InitializeComponent();
+        app = this;
+
+#if defined _DEBUG && !defined DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION
+        UnhandledException([](winrt::IInspectable const&, winrt::UnhandledExceptionEventArgs const& e)
+            {
+                if (IsDebuggerPresent())
+                {
+                    auto errorMessage = e.Message();
+                    __debugbreak();
+                }
+            });
+#endif
+    }
+
+    void App::ToForeground()
+    {
+        assert(app != nullptr);
+
+        HWND hwnd;
+        auto WindowNative{ app->window.as<IWindowNative>() };
+        if (WindowNative != nullptr && WindowNative->get_WindowHandle(&hwnd) == S_OK)
+        {
+            SwitchToThisWindow(hwnd, TRUE);
+        }
+    }
+
+    std::wstring App::GetFullPathToExe()
+    {
+        TCHAR buffer[MAX_PATH] = { 0 };
+        GetModuleFileName(NULL, buffer, MAX_PATH);
+        std::wstring::size_type pos = std::wstring(buffer).find_last_of(L"\\/");
+        return std::wstring(buffer).substr(0, pos);
+    }
+
+    std::wstring App::GetFullPathToAsset(std::wstring const& assetName)
+    {
+        return GetFullPathToExe() + L"\\Assets\\" + assetName;
+    }
+
+    void App::OnLaunched(winrt::Microsoft::UI::Xaml::LaunchActivatedEventArgs const& /*args*/)
+    {
+        window = make<MainWindow>();
+
+        g_notificationManager.Init();
+
+        // NOTE: AppInstance is ambiguous between
+        // Microsoft.Windows.AppLifecycle.AppInstance and
+        // Windows.ApplicationModel.AppInstance
+        auto currentInstance{ winrt::AppInstance::GetCurrent() };
+        if (currentInstance)
+        {
+            // AppInstance.GetActivatedEventArgs will report the correct ActivationKind,
+            // even in WinUI's OnLaunched.
+            winrt::AppActivationArguments activationArgs{ currentInstance.GetActivatedEventArgs() };
+            if (activationArgs)
```

Comment:
Is this check needed. I think we are guaranteed an activationArgs of Launch at the minimal every single time the app launches.